### ISSUE TITLE
crash on start without notification demon

### ISSUE THREAD

**Issue by wanderer** [ssbc/patchwork#378]
93bdf60b9efb60494a31549e761d3511a2da029c causing patchwork to crash on start if the computer doesn't have a notification demon installed

**Comment by kyrias** [comment]
It sounds like this should maybe be handled in `ssb-notifier` instead though.

**Comment by clehner** [comment]
@wanderer is there a stack trace when it crashes?

**Comment by wanderer** [comment]
no it just does this
```
> ssb-patchwork@2.7.0 start /home/null/code/patchwork
> node ./index.js

[CFG] Password: NO.
[CFG] TLS: NO.
[CFG] Remote Access: NO.
Patchwork - Copyright (C) 2015-2016 Secure Scuttlebutt Consortium
This program comes with ABSOLUTELY NO WARRANTY.
This is free software, and you are welcome to redistribute it under certain conditions (GPL-3.0).

Starting...
Loading plugins from /home/null/.ssb/node_modules
Log level: notice
Serving at http://localhost:7777
Process org.freedesktop.Notifications exited with status 1

npm ERR! Linux 4.5.1-1-ARCH
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "start"
npm ERR! node v6.0.0
npm ERR! npm  v3.8.8
npm ERR! code ELIFECYCLE
npm ERR! ssb-patchwork@2.7.0 start: `node ./index.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the ssb-patchwork@2.7.0 start script 'node ./index.js'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the ssb-patchwork package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node ./index.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs ssb-patchwork
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls ssb-patchwork
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/null/code/patchwork/npm-debug.log
```

**Comment by wanderer** [comment]
works fine after installed the notication demon

**Comment by pfraze** [comment]
@wanderer side-question, is your user's name 'null'? Just want to make sure that's not another bug

**Comment by clehner** [comment]
This will need some more debugging to find how to handle the error in ssb-notifier or freedesktop-notifications

**Comment by wanderer** [comment]
@pfraze yes im null.

**Comment by Kubuxu** [comment]
I am getting this error:
```
The name org.freedesktop.Notifications was not provided by any .service files
```

Which means that there is nothing providing Notifications daemon on my Linux installation.

Notification daemon should be required to run patchwork.

**Comment by clehner** [comment]
> Notification daemon should be required to run patchwork.

We got by without them for a while - I think it should just fail gracefully and not show notifications if there is no notification daemon.

What packages do I need to uninstall from my system to reproduce this issue?

**Comment by Kubuxu** [comment]
(There should be `not` in my previous comment)

Are you using systemd?

**Comment by clehner** [comment]
> (There should be `not` in my previous comment)

Ok

> Are you using systemd?

Yes, on debian testing

**Comment by Kubuxu** [comment]
You can check with: `grep -R Notification /usr/share/dbus-1/{,system-}services` move the file that contains information about Notification bus, **restart**, you should be good to test.

**Comment by clehner** [comment]
@Kubuxu thanks. I found two files in there, one from xfce4-notifyd and one from dunst. I didn't feel like restarting so I uninstalled the packages and then was able to reproduce the issue. I'm patching ssb-notifier now to fix the crash

**Comment by clehner** [comment]
Fixed in ssb-notifier 2.0.1. Update with `cd patchwork; npm update ssb-notifier`
